[lib & admin & template  & plugins] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8')

zero-24 opened the pull request:
Pull Request for Issue #10399 .
#### Summary of Changes

Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8')
#### Testing Instructions

Please do a code review or test all changed files.

zero-24 committed:
Update highlight.php

zero-24 committed:
Update default.php

zero-24 committed:
Update view.html.php

zero-24 committed:
Update error.php

zero-24 committed:
Update default_items.php

zero-24 committed:
Update offline.php

zero-24 committed:
Update opensearch.php

zero-24 committed:
Update debug.php

zero-24 committed:
Update component.php

brianteeman commented:
I have tested this item :white_check_mark: successfully on 6aa2082258b0a1ff586f4343fb6387ede2cfcd3e<br /><br />on review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

alikon commented:
I have tested this item :white_check_mark: successfully on 6aa2082258b0a1ff586f4343fb6387ede2cfcd3e<br /><br />on review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

brianteeman commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

andrepereiradasilva commented on libraries/joomla/document/opensearch.php:
```diff
@@ -133,11 +133,11 @@ public function render($cache = false, $params = array())
 
 		$elShortName = $xml->createElementNs($osns, 'ShortName');
-		$elShortName->appendChild($xml->createTextNode(htmlspecialchars($this->_shortName)));
+		$elShortName->appendChild($xml->createTextNode(htmlspecialchars($this->_shortName), ENT_COMPAT, 'UTF-8'));
```
this is not correct!

brianteeman replied:
Why not?

andrepereiradasilva replied:
´$xml->createTextNode(htmlspecialchars($this->_shortName), ENT_COMPAT, 'UTF-8')`

should be

`$xml->createTextNode(htmlspecialchars($this->_shortName, ENT_COMPAT, 'UTF-8'))`

andrepereiradasilva commented on libraries/joomla/document/opensearch.php:
```diff
@@ -133,11 +133,11 @@ public function render($cache = false, $params = array())
 
 		$elDescription = $xml->createElementNs($osns, 'Description');
-		$elDescription->appendChild($xml->createTextNode(htmlspecialchars($this->description)));
+		$elDescription->appendChild($xml->createTextNode(htmlspecialchars($this->description), ENT_COMPAT, 'UTF-8'));
```
this is not correct!

andrepereiradasilva commented on libraries/joomla/document/opensearch.php:
```diff
@@ -151,7 +151,7 @@ public function render($cache = false, $params = array())
 			$elImage->setAttribute('width', $image->width);
 			$elImage->setAttribute('height', $image->height);
-			$elImage->appendChild($xml->createTextNode(htmlspecialchars($image->data)));
+			$elImage->appendChild($xml->createTextNode(htmlspecialchars($image->data), ENT_COMPAT, 'UTF-8'));
```
this is not correct!

andrepereiradasilva commented:
@brianteeman please remove the RTC from this one. It as errors.

brianteeman commented:
Please explain the errors - saying it is not correct is not helpful

andrepereiradasilva commented:
`htmlspecialchars($image->data), ENT_COMPAT, 'UTF-8'`

brianteeman commented:
Explain not quote ;)

andrepereiradasilva commented:
the parentisis is the wrong place

andrepereiradasilva commented:
travis is also telling that.

brianteeman commented:
RTC removed as requested<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

wojsmol commented:
@zero-24 see https://github.com/zero-24/joomla-cms/issues/12

wojsmol committed:
fix for travis (#12)

* fix travis error

* CS

andrepereiradasilva commented:
seems all fine to me now.

zero-24 committed:
Merge branch 'staging' into patch-33

zero-24 commented:
Thanks @wojsmol & @andrepereiradasilva

Commit:
Fix the URL for technical requirements on joomla.org (#10461)

Fix the URL for technical requirements on joomla.org
following:
https://github.com/joomla/joomla-websites/issues/333 and https://github.com/joomla/joomla-cms/pull/10446#issuecomment-219108807

**Testing instruction:**

Easy fix on review

wojsmol committed:
Code style  changes for plg_system_sef (#10465)

* Code style chenges for plg_system_sef

* extra space

wojsmol committed:
[plg_content_joomla]Code style (#10470)

* [plg_content_joomla]Code style

* tabs

wojsmol committed:
[plg_authentication_gmail] Code Style (#10467)

* Code Style

* remove blank lines

wojsmol committed:
[plg_authentication_ldap] XML code style (#10468)

wojsmol committed:
[plg_content_emailcloak]XML code style (#10469)

brianteeman committed:
Update en-AU installation file (#10477)

Commit:
Updated installation language files

brianteeman committed:
English ca (#10478)

brianteeman committed:
Update US installation (#10479)

brianteeman commented:
I have tested this item :white_check_mark: successfully on 22ff770e35d41f7b1c6309b1efc013c43d9d426b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

andrepereiradasilva commented:
I have tested this item :white_check_mark: successfully on 22ff770e35d41f7b1c6309b1efc013c43d9d426b<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

wojsmol committed:
[plg_editors_tinymce]Code style (#10484)

* [plg_editors_tinymce]Code style

* options in one line

brianteeman commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

brianteeman committed:
correct technical requirements link (#10514)

brianteeman committed:
Update en-GB.com_finder.ini (redo 7879) (#10513)

* redo 7879

* singular

wojsmol committed:
[plg_user_profile]Code style (#10511)

wojsmol committed:
[plg_user_joomla]Code style (#10510)

* [plg_user_joomla]Code style

* cs

wojsmol committed:
[plg_user_contactcreator]XML code style (#10509)

wojsmol committed:
[plg_twofactorauth_yubikey]Code style (#10508)

* [plg_twofactorauth_yubikey]Code style

* indent

wojsmol committed:
[plg_twofactorauth_totp]XML code style (#10507)

wojsmol committed:
[plg_system_updatenotification]Code style (#10506)

* [plg_system_updatenotification]Code style

* cs

wojsmol committed:
[plg_system_stats]Code style (#10505)

wojsmol committed:
[plg_system_redirect]XML code style (#10503)

wojsmol committed:
[plg_system_p3p]XML code style (#10502)

wojsmol committed:
[plg_system_log]XML code style (#10501)

wojsmol committed:
[plg_system_languagefilter]Code style (#10500)

* [plg_system_languagefilter]Code style

* new line

wojsmol committed:
[plg_system_languagecode]Code style (#10499)

wojsmol committed:
[plg_system_highlight]Code style (#10498)

wojsmol committed:
[plg_system_debug]Code style (#10497)

* [plg_system_debug]Code style

* cs

wojsmol committed:
[plg_system_cache]Code style (#10496)

wojsmol committed:
[plg_search_tags]Code style (#10495)

* [plg_search_tags]Code style

* more cs

* cs

wojsmol committed:
[plg_search_newsfeeds]Code style (#10494)

* [plg_search_content]Code style

* [plg_search_newsfeeds]Code style

* Revert "[plg_search_content]Code style"

This reverts commit 41f564c8f944ff2bad36d0ef690b528ec47a8440.

* remove blank line

wojsmol committed:
[plg_search_content]Code style (#10493)

wojsmol committed:
[plg_search_contacts]Code style (#10492)

wojsmol committed:
[plg_search_categories]Code style (#10491)

wojsmol committed:
[plg_quickicon_joomlaupdate]XML code style (#10490)

wojsmol committed:
[plg_quickicon_extensionupdate]XML code style (#10489)

wojsmol committed:
[plg_editors-xtd_readmore]Code style (#10487)

wojsmol committed:
[plg_editors-xtd_pagebreak]Code style (#10486)

wojsmol committed:
[plg_editors-xtd_module]Code style (#10485)

wojsmol committed:
[plg_editors_none]code style (#10483)

brianteeman committed:
copyright header (#10476)

wojsmol committed:
[plg_content_pagenavigation]Code style (#10474)

* [plg_content_pagenavigation]Code style

* XML CS

* options in one line

* indent

wojsmol committed:
[plg_content_pagebreak]Code style (#10473)

* [plg_content_pagebreak]Code style

* more XML CS

* options on one line + indent

wojsmol committed:
[plg_content_loadmodule]XML code style (#10472)

* [plg_content_loadmodule]XML code style

* XML CS

* options in one line

wojsmol committed:
[plg_captcha_recaptcha]code style (#10471)

* [plg_captcha_recaptcha]code style

* indent

* indent

* XML CS

* options in one line

Commit:
Remove obsolte bootstrap and jquery files in ISIS template (#10445)

brianteeman committed:
Update INSTALL (#10446)

cyrez committed:
Modal com_contact: improve Select & Edit Contact (#10457)

* Improve select contact modal

* Add Edit contact modal + simplify and clean code

* Language string consistency: COM_CONTACT_EDIT_CONTACT

« Edit Contact » as for other components: com_content, com_categories…

zero-24 committed:
[admin & frontend & modules] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8') (#10443)

* Update index.php

* Update login.php

* Update manager.php

* Update view.html.php

* Update blog.php

* Update default_items.php

* Update default.php

zero-24 committed:
[mod_* frontend & protostar] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8') (#10417)

* Update mod_articles_archive.php

* Update mod_articles_latest.php

* Update mod_articles_news.php

* Update mod_articles_popular.php

* Update mod_languages.php

* Update mod_languages.php

* Update mod_stats.php

* Update modules.php

* Update helper.php

zero-24 committed:
Update mod_stats_admin.php (#10415)

zero-24 committed:
Update mod_users_latest.php (#10414)

zero-24 committed:
Update mod_custom.php (#10413)

zero-24 committed:
Update title.php (#10412)

zero-24 committed:
Update default.php (#10411)

zero-24 committed:
Update modules.php (#10410)

zero-24 committed:
Update default.php (#10409)

zero-24 committed:
Update modules.php (#10408)

zero-24 committed:
Update mod_syndicate.php (#10407)

zero-24 committed:
Update mod_search.php (#10405)

zero-24 committed:
Update intro_image.php (#10404)

zero-24 committed:
[lib/mod/templates] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8') (#10430)

* Update head.php

* Update modules.php

* Update offline.php

* Update default.php

* Update vote.php

andrepereiradasilva committed:
error red color for error, green message for success (#10425)

zero-24 committed:
[mod_* templates] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8') (#10422)

* Update mod_footer.php

* Update mod_whosonline.php

* Update mod_feed.php

* Update mod_banners.php

* Update mod_breadcrumbs.php

* Update mod_feed.php

* Update mod_random_image.php

* Update mod_related_items.php

* Update mod_tags_similar.php

* Update mod_menu.php

* Update error.php

* Update users.php

zero-24 committed:
Update default_items.php (#10420)

zero-24 committed:
Update default_links.php (#10403)

zero-24 committed:
Update default_links.php (#10402)

zero-24 committed:
[beez3 & other] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8') (#10431)

* Update languages.php

* Update cloud.php

* Update default.php

* Update blog_item.php

* Update default_item.php

* Update default.php

* Update default.php

* Update default_system.php

* fix php error (#14)

kavadas committed:
Proper alias transliteration based on the content language (#10348)

* Added proper transliteration mechanism

* Code updated to get the correct language based on the tag provided

* Fixed banners alias transliteration issue

* Updated banners alias generation to follow the same pattern as the rest core extensions.

* Moved the site language detection from JFilterOutput::stringURLSafe to JApplicationHelper::stringURLSafe

* Newsfeed model also contains code for alias generation. Updated that code to transliterate properly.

zero-24 committed:
Codestyle installation views (#10359)

* codestlye installation

* codestlye installation2

* remove to much spaces

zero-24 committed:
Update mod_wrapper.php (#10401)

andrepereiradasilva committed:
wrap lines in json error (#10367)

zero-24 committed:
[comp & lib] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8') (#10437)

* Update default_items.php

* Update list.php

* Update view.html.php

* Update view.html.php

* Update view.html.php

* Update route.php

* Update mysqli.php

* Update oauth.php

* Update mod_articles_category.php

* Update mysqli.php

rdeutz commented:
@zero-24 could you have a look at the merge conflicts, thanks.

matrikular committed:
Change scheme for the google maps api call to https and include an api key if one was set (#10439)

* Change api scheme to https

While the main change to this file is about setting the scheme to https, I've also refactored the uri handling and added the option to apply an api key as well.

* Remove urlencode from address since it will be handled automatically in toString()

* Create a new JUri instance instead of using $this->uri to prevent errors with e.g. isSercure()

* Remove the sensor parameter since it doesn't seem to be need any longer

Remove the sensor parameter since it doesn't seem to be need any longer. Thanks for pointing it out @robert.

* Satisfy Travis, maybe?!

* Add the sensor parameter back to the api url since the tests seem to rely on it

Add the sensor parameter back to the api url since the tests seem to rely on it. If that really is the case, we should implement be a better way of "parsing" the url and / or looking for an array index in the tests.

* Fix url parsing to properly handle query params

* Remove the sensor parameter since it doesn't seem to be needed any more

This time the test(s) shouldn't fail - fingers crossed.

zero-24 committed:
[tags & users & libs & beez] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8') (#10440)

* Update default.php

* Update category_default.php

* Update pdomysql.php

* Update object.php

* Update default_items.php

* Update default_items.php

* Update default.php

* Update view.html.php

* Update categories.php

* codestyle https://travis-ci.org/joomla/joomla-cms/jobs/129754482

cyrez committed:
Modal com_categories: improve Select & Edit Category (#10441)

* Improve categories modal container

* Add modal viewport dimensions

* Remove single "double quote" in token

* Fix JS error Uncaught SyntaxError: Unexpected token )

* Array indent

* Add edit category as a modal and other improvements

csthomas committed:
Do not count all table rows if start offset is 0 (#9162)

cyrez committed:
Check out in case of modal close not needed anymore (set in bootstrap.renderModal) (#10428)

izharaazmi committed:
Use getState instead of direct property read (#10435)

* Use getState instead of direct property read

Use getState instead of direct property read to avoid any undefined property warnings.

* Fix test case

We're setting state manually for this test case. But since the flag is not set the populateState will be called and it will overwrite them.

cyrez committed:
[com_categories] Coding standards xml + fix a wrong closing tag (#10515)

* Code style filter_categories.xml

Add empty line before opening tags for readability

* Code style category.xml + fix closing tag field "tags"

* Remove extra empty lines at end of categories.xml

cyrez committed:
Modal com_newsfeeds: improve Select & Edit Newsfeed (#10516)

* Simplify newsfeed modal, by setting edit layout

* Improve Select & Edit com_newsfeeds modals

andrepereiradasilva committed:
[js message system] Add close button (#10517)

* add close button to js messages

* type button

andrepereiradasilva committed:
correct js messages class (#10521)

andrepereiradasilva committed:
[com_categories] Cleaner tree view (#10213)

* cleaner tree

* new view

* and now?

andrepereiradasilva committed:
use the same level structure as categories

andrepereiradasilva committed:
add to com_config filters and solve html error (th to td)

infograf768 committed:
[imp] New lang string for successful language installation (#10523)

* New lang string for successful language installation

* took off pack

Kubik-Rubik committed:
Merge pull request #10526 from andrepereiradasilva/patch-7

[com_tags] use the same visual level structure as categories

andrepereiradasilva committed:
[com_users] user groups: use the same visual level structure as categories (#10528)

* user groups tree

* forgot the *

* add space

* remove space

andrepereiradasilva committed:
debuguser and debuggroup tree view (#10530)

Kubik-Rubik committed:
Merge pull request #10531 from andrepereiradasilva/patch-13

[com_config] filters tab: use the same tree level visual

andrepereiradasilva committed:
use the new layout (#10536)

andrepereiradasilva committed:
tree prefix for rules field. (#10532)

andrepereiradasilva committed:
[com_categories] modal: use the same tree level visual (#10533)

* use layout

* add tree prefix to categories modal

zero-24 committed:
merge conflicts

zero-24 commented:
i did broken my branch please remove the milestone. @brianteeman
